### PR TITLE
OS-593: refactors repo for linting and linting during workflows

### PR DESCRIPTION
**Pull request by mathewmeconry**

### CODE REVIEW

File: .github/workflows/formatting-linting.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          cache: 'yarn'
+          node-version: 16
+
+      - name: 'Install the dependencies'
+        run: 'yarn install'
+
+      - name: 'Check code formatting'
+        run: yarn prettier:check
+
+      - name: 'Lint contracts solitidy'
```
Comment (heueristik) [inline .github/workflows/formatting-linting.yml]:
```suggestion
      - name: 'Lint contracts solidity'
```

File: packages/js-client/package.json [hunk 2]
```diff
@@ -64,7 +57,7 @@
     "@aragon/osx-ethers": "1.3.0-rc0.3",
     "@aragon/sdk-client-common": "1.2.0-rc0",
     "@aragon/sdk-common": "1.5.0",
-    "@aragon/simple-storage-ethers": "1.0.0",
+    "@aragon/simple-storage-ethers": "file:../contracts-ethers",
```
Comment (heueristik) [inline packages/js-client/package.json:60]: Oh, this should not be needed in this repo. Any reason why you added it @josemarinas or was this by accident?

File: packages/js-client/package.json [hunk 3]
```diff
@@ -74,6 +67,9 @@
     "graphql": "^16.6.0",
     "graphql-request": "4.3.0"
   },
+  "bundledDependencies": [
+    "@aragon/simple-storage-ethers"
+  ],
```
Comment (heueristik) [inline packages/js-client/package.json:72]: see above

File: packages/js-client/package.json (hunk 2 above)
Comment (heueristik) [reply packages/js-client/package.json:60]: Ah, I think the naming in `aragon/osx-plugin-template-hardhat/packages/contracts-ethers/package.json`
```
{
  "name": "@aragon/simple-storage-ethers",
  "version": "1.0.0",
  "description": "Plugin contract definitions for ethers.js",
  "main": "dist/bundle-cjs.js",
  "module": "dist/bundle-esm.js",
  "types": "dist/bundle.d.ts",
```
is wrong. Please fix it in another PR @josemarinas. I created a task for it https://aragonassociation.atlassian.net/browse/OS-781

**Review by heueristik — CHANGES_REQUESTED** [review]
Awesome!
Just found a typo and wrong naming of the `js-client` package that @josemarinas can fix.

**Review by heueristik — APPROVED** [review]

**Review by Rekard0 — APPROVED** [review]
nice

**Pull request merged**